core/util/google.py

```diff
@@ -70,14 +70,15 @@ def run_crawl(self):
 			payload['start'] = set_page(page)
 			if page >= self.limit:
 				break
-		links = self.framework.page_parse(self._pages).findall(r'a href="([^"]+)"')
+		links = self.framework.page_parse(self._pages).findall(r'a href=".*?(https?[^"]+)"')
```

Comment:
.*? is a dangerous expr. Instead of this, we need a specific regex.
Comment a demo URL so we can discuss.

Reply:
/url?q=https://www.khanacademy.org/computing/computer-programming/programming/intro-to-programming/v/programming-intro&amp;sa=U&amp;ved=2ahUKEwiovqTyp5fvAhWf63MBHaStCCAQuAIwDnoECAQQAg&amp;usg=AOvVaw19d7CnNCKqZQJe8sEy-shq

Reply:
re.sub(r"^/url\?q=", '', link)
for spliting the url, use more stable chars: &amp;ved=..